platforms/core-runtime/base-services/src/main/java/org/gradle/initialization/BuildCancellationToken.java

```diff
@@ -16,9 +16,13 @@
 
 package org.gradle.initialization;
 
+import org.gradle.internal.service.scopes.Scope;
+import org.gradle.internal.service.scopes.ServiceScope;
+
 /**
  * Propagates notification that the build should be cancelled.
  */
+@ServiceScope({Scope.Global.class, Scope.Build.class, Scope.BuildSession.class})
```

Comment:
❓ Does this need all the scopes listed? Why not just `Global`?

Reply:
Oh, I actually I can't remove `Build` it fails without it